src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -532,25 +551,12 @@ function VisualizeAppController(
         timefilter.setRefreshInterval(savedQuery.attributes.timefilter.refreshInterval);
       }
     }
-
-    $scope.fetch();
   };
 
-  // update the query if savedQuery is stored
-  if (stateContainer.getState().savedQuery) {
-    savedQueryService.getSavedQuery(stateContainer.getState().savedQuery).then(savedQuery => {
-      $scope.$evalAsync(() => {
-        $scope.savedQuery = savedQuery;
-      });
-    });
-  }
-
-  $scope.$watch('savedQuery', newSavedQuery => {
-    if (!newSavedQuery) return;
-    stateContainer.transitions.set('savedQuery', newSavedQuery.id);
-
-    updateStateFromSavedQuery(newSavedQuery);
-  });
+  $scope.updateSavedQuery = savedQuery => {
+    $scope.savedQuery = savedQuery;
```

Comment:
Why do you still need to store the savedQuery on the scope?
I think you could get rid of this (see discover)

Reply:
It's used in `<kbn-top-nav>`.